pkg/apis/resource/types.go

```diff
@@ -155,8 +162,31 @@ type AllocationResult struct {
 	Shareable bool
 }
 
-// ResourceHandleMaxSize is the maximum size of allocation.resourceHandle.
-const ResourceHandleMaxSize = 16 * 1024
+// AllocationResultContextMaxSize represents the maximum number of entries in
+// allocation.context.
+const AllocationResultContextMaxSize = 32
+
+// ResourceHandle holds opaque resource data for processing by a specific kubelet plugin.
+type ResourceHandle struct {
+	// KubeletPluginName specifies the name of the kubelet plugin which should
+	// be invoked to process this ResourceHandle's data. If this field is not
+	// set, the kubelet plugin with the same name as the DriverName from the
+	// ResourceClaimStatus this ResourceHandle is embedded in will be invoked.
+	// +optional
+	KubeletPluginName string
```

Comment:
KubeletPluginName vs DriverName ?  I'm not seeing why the field names are different - it feels like `claim.status.context[x].driverName` means the same thing as `claim.status.driverName`, no?

Why are they named differently?  I prefer to avoid embedding things like "kubelet" into the API schema

Reply:
Updated